lib/infra/infra-stack.ts

```diff
@@ -79,35 +82,71 @@ export class InfraStack extends Stack {
     const ec2InstanceType = (props.cpuType === AmazonLinuxCpuType.X86_64)
       ? InstanceType.of(InstanceClass.C5, InstanceSize.XLARGE) : InstanceType.of(InstanceClass.C6G, InstanceSize.XLARGE);
 
+    const alb = new NetworkLoadBalancer(this, 'publicNlb', {
```

Comment:
Why NLB?
Nit: variable says ALB and it is using NLB maybe fix the var name.

Reply:
NLB for handling high amount of concurrent requests, will update the variable name.